accumulator/undo.go

```diff
@@ -34,8 +36,105 @@ func (u *undoBlock) ToString() string {
 	return s
 }
 
+// SerializeSize returns how many bytes it would take to serialize this undoblock.
+func (u *UndoBlock) SerializeSize() int {
+	// Size of u.numAdds + len(u.positions) + each position takes up 8 bytes
+	size := 4 + 8 + (len(u.positions) * 8)
+
+	// Size of len(u.hashes) + each hash takes up 32 bytes
+	size += 8 + (len(u.hashes) * 32)
+
+	return size
+}
+
+// Serialize encodes the undoblock into the given writer.
+func (u *UndoBlock) Serialize(w io.Writer) error {
+	err := binary.Write(w, binary.BigEndian, u.numAdds)
+	if err != nil {
+		return err
+	}
+
+	err = binary.Write(w, binary.BigEndian, uint64(len(u.positions)))
+	if err != nil {
+		return err
+	}
+
+	for _, pos := range u.positions {
+		err := binary.Write(w, binary.BigEndian, pos)
+		if err != nil {
+			return err
+		}
+
+	}
```

Comment:
It should be possible to do this:
```suggestion
	err = binary.Write(w, binary.BigEndian, u.positions)
	if err != nil {
		return err
	}
```

https://github.com/golang/go/blob/6e738868a7a943d7d4fd6bb1963e7f6d78111726/src/encoding/binary/binary.go#L354

Reply:
Change applied